src/plugin/retail.js

```diff
@@ -0,0 +1,32 @@
+var logger = require('../utils/utils').logger;
+
+var preventKeysList = [];
+
+function shouldPreventKeyDefaultBehavior(key) {
+    if (!preventKeysList.length) return false;
+
+    return !!preventKeysList.find(function(preventKey) {
+        return preventKey.key === key;
+    });
+}
+
+function Retail() {}
+
+Retail.prototype.setUpOnce = function() {
+    window.document.addEventListener('keydown', function(e) {
+        if (shouldPreventKeyDefaultBehavior(e.key)) {
+            e.preventDefault();
+            logger.out('debug', '[GGP] Plugin Retail:  Preventing key default behavior.', {
+                key: e.key
+            });
+        }
+    });
+};
+
+Retail.prototype.onLoad = function(slave, loadData) {
+    loadData.data.settings.preventKeys.forEach(function(key) {
```

Comment:
If `preventKeys` is not sent for some reason this will throw error. @thedev966

Reply:
@dikaso It's possible case? Okay, I'll add some check in this next task I'm on.

Reply:
I think so. It is not required to set `preventKeys` for a tenant.

Reply:
But, we will always set `preventKeys`, by default empty array, and override settings per tenant when needed. `forEach` shouldn't throw any error..